src/CmdLine.hs

```diff
@@ -140,6 +141,14 @@ data Cmd
         ,cmdCppSimple :: Bool
         ,cmdCppAnsi :: Bool
         }
+    | CmdSummary
+        {cmdGivenHints :: [FilePath]
+        ,cmdWithGroups :: [String]
+        ,cmdIgnore :: [String]
+        ,cmdFindHints :: [FilePath]
+        ,cmdNoImplicit :: Bool           -- ^ Do not use .hlint.yaml implicitly
```

Comment:
What precisely does "use .hlint.yaml implicitly" mean?

Reply:
If no explicit hint file is specified via `--hint`, HLint looks for `.hlint.yaml` in the current directory and uses it implicitly. This flags disables that, which is useful when running `cabal run hlint summary` from HLint dir, because HLint's `.hlint.yaml` is only relevant to the HLint codebase.

Reply:
Ok. So, if not disabled with `--no-implicit`, the `.hlint.yaml` file found in the project will be consulted and any hints it contains will be reported on. We generally don't want this because such rules are specific to the hlint project itself  (not for general users of hlint). Have I got that right?

Reply:
That's exactly right. It's main purpose is to avoid using HLint's `.hlint.yaml` when generating `hints.md`. When users run it on their own repos they most likely do want to include their `.hlint.yaml` files.